api/app/src/main/kotlin/packit/service/UserService.kt

```diff
@@ -0,0 +1,49 @@
+package packit.service
+
+import org.springframework.stereotype.Service
+import packit.model.User
+import packit.model.UserGroup
+import packit.repository.UserGroupRepository
+import packit.repository.UserRepository
+import packit.security.Role
+import java.time.Instant
+
+interface UserService
+{
+    fun saveUserFromGithub(username: String, displayName: String?, email: String?): User
+    fun getUserRoleUserGroup(): UserGroup
+    fun getAdminRoleUserGroup(): UserGroup
+}
+
+@Service
```

Comment:
I think it would be nice to put some comment on these various service classes about what concerns each of them has. I'm not completely clear why this is "Base"user service - it seems just be some useful utils over the repository class?

Reply:
ive updated naming just to make the interface and main class distinct... but its just the usual controller --> service --> repository pattern... don't need comments imo